### ISSUE TITLE
average_partial_effect gets "NA" when there is no missing value in tau.forest$predictions

### ISSUE THREAD

**Issue by ZhangMengxia** [grf-labs/grf#785]
Dear authors,

I encounter a problem when estimating the effect of whether the unit's age<1.

After running the following code
`tau.forest = causal_forest(X, Y, W,  num.threads = 5, seed=1234, clusters=clusters, equalize.cluster.weights=FALSE,  min.node.size=5)`
`result=average_partial_effect(tau.forest)`

The result is 
**estimate  std.err 
     NaN      NaN** 

I have checked `tau.forest$predictions`, there is no missing value in the predictions. I wonder what else I should check and how to solve this problem? 

Thank you very much and happy new year!

**Comment by erikcs** [comment]
When the treatment is continuous GRF estimates the conditional variance of W needed for the AIPW scores using a [regression_forest](https://github.com/grf-labs/grf/blob/master/r-package/grf/R/get_scores.R#L74), if some of these predictions are zero then NaNs will arise from a division by zero.

That seems unlikely to happen, do you have a suggestion here @swager ?

**Comment by erikcs** [comment]
Also, @ZhangMengxia, could you please try your example with the development version of grf, and use `average_treatment_effect` instead (you can install it with `devtools::install_github("grf-labs/grf", subdir = "r-package/grf")`) - there have been some changes to the summary functions (#723) since the last CRAN version.

**Comment by ZhangMengxia** [comment]
@erikcs Thanks!! 

I have a question before I try the development version. I know that the causal forest first uses a regression forest to predict W and Y and retain the residuals. After this step, the variable indicating whether the unit's age<1 is no longer binary but instead continuous. Is it still okay to use "average_treatment_effect" function?  Many thanks in advance!

**Comment by erikcs** [comment]
In the development version `average_treatment_effect` is the main entry to ate/cape/etc, everything was just moved to one function.

**Comment by ZhangMengxia** [comment]
@erikcs Thanks a lot.

First, I have tried the `average_treatment_effect` within `grf_1.2.0`,  the result becomes normal (i.e., not NaN). There is also some warning message:
```
Warning message:
In average_treatment_effect(tau.forest) :
  Estimated treatment propensities go as low as 0 which means that treatment effects for some controls may not be well identified. In this case, using `target.sample=treated` may be helpful.
```
I wanted to try the `average_treatment_effect` from `devtools::install_github("grf-labs/grf", subdir = "r-package/grf")`, but I did not figure out how to install devtools on my Linux machine. The information [here](https://www.r-project.org/nosvn/pandoc/devtools.html) for Linux is not very clear. Sorry :( 

Second, I got a question. Does this `average_treatment_effect` function use the doubly robust corrections (i.e., AIPW scores) when summarizing the average treatment effect? 

Thanks!!

**Comment by erikcs** [comment]
You can install it with `install.packages("devtools")`

**Comment by ZhangMengxia** [comment]
@erikcs I was finally able to install both "devtools" and the "grf" in development. To make sure I installed the right version, should it be `grf_1.2.0.0`?

I tried both `average_partial_effect` and `average_treatment_effect` from `grf_1.2.0.0`, the results are shown in the following screenshot.
![Screen Shot 2021-01-06 at 8 44 50 PM](https://user-images.githubusercontent.com/17790770/103852109-1091b880-5060-11eb-820b-75472ea98913.png)

Then I follow the suggestion and results are below

![Screen Shot 2021-01-06 at 8 52 08 PM](https://user-images.githubusercontent.com/17790770/103852652-2bb0f800-5061-11eb-99a6-d218959e3406.png)

I tried to check the code of `average_treatment_effect` [here](https://github.com/grf-labs/grf/blob/5a4f07cdc517f996eca0465deec3ad75c023985c/r-package/grf/R/average_treatment_effect.R#L201). Does it mean the new version of `average_treatment_effect` reports 'NaN' because it uses AIPW?

**Comment by erikcs** [comment]
Thanks! 

The new version warns that the treatment propensities are zero, so I'd say this problem is resolved by #723.

>I tried to check the code of average_treatment_effect here. Does it mean the new version of average_treatment_effect reports 'NaN' because it uses AIPW? And why would the old version of average_treatment_effect does not report 'NaN' for ATE?

You can see the details here:
https://grf-labs.github.io/grf/reference/average_treatment_effect.html#details

**Comment by ZhangMengxia** [comment]
@erikcs I have another related question. I happened to use the two summary functions 'average_partial_effect' and 'average_treatment_effect' for a binary variable, but the results are very different. Do you have any ideas about why it is the case? Thanks!

![Screen Shot 2021-01-07 at 5 11 55 PM](https://user-images.githubusercontent.com/17790770/103962385-b2baaa80-510b-11eb-9cf8-5fece568a61f.png)

**Comment by erikcs** [comment]
Hi @ZhangMengxia, they perform different calculations and may differ differ. In your case you have a problem with zero overlap as many of your estimated propensities are exactly zero - that may exacerbate the differences between the two

**Comment by erikcs** [comment]
Closing as resolved by #723, please feel free to re-open if anything is unclear.
